File: integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/model/CarEntity.java
package io.quarkus.it.mongodb.panache.model;

import io.quarkus.mongodb.panache.common.MongoEntity;

@MongoEntity
public class CarEntity extends VehicleEntity {

    public CarEntity() {
    }

    public CarEntity(String id, Integer modelYear) {
        super(id, modelYear);
    }
}

File: integration-tests/mongodb-panache-kotlin/src/test/kotlin/io/quarkus/it/mongodb/panache/model/CarEntity.kt
package io.quarkus.it.mongodb.panache.model

import io.quarkus.mongodb.panache.common.MongoEntity
import org.bson.codecs.pojo.annotations.BsonCreator
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.codecs.pojo.annotations.BsonProperty

@MongoEntity
data class CarEntity @BsonCreator constructor(
    @BsonId override var id: String,
    @BsonProperty("modelYear") override var modelYear: Int?
) : VehicleEntity(id, modelYear)
